### PR TITLE
Nodejs week2

### PR DESCRIPTION
**Pull request by Aditi-Sawardekar**
Initial Commit

### CODE REVIEW

File: nodejs/week2/app.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+const express = require("express");
+const app = express();
+const port = process.env.PORT || 3000;
+
```
Comment (r0hitsharma) [inline nodejs/week2/app.js]: This is fine, but please try and use es modules.

File: nodejs/week2/app.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+app.post("/search", (req, res) => {
+    const input = (req.query.q);
+    const [field] = Object.values(req.body);
+    
+    try {
+        if (input) {
+            const search = input.toLowerCase();
+
+            const filteredItems = documents.filter((item) => {
+                const result = ([Object.values(item)].toString().includes(search));
```
Comment (r0hitsharma) [inline nodejs/week2/app.js]: Could you think of a better way to do this filtering?

File: nodejs/week2/app.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            res.send(filteredItems)
+        } else if (field) {
+            const [search] = Object.entries(field)
+            
+            const filteredItems = documents.filter((item) => {
+                const result = ([Object.entries(item)].toString().includes(search));
+                return (result);
+            });
+
+            res.send(filteredItems)
```
Comment (r0hitsharma) [inline nodejs/week2/app.js]: What if filtered items are empty?

File: nodejs/week2/app.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        } else {
+            res.send(documents)
+        }
+    } catch (error) {
+        return res.send(error);
```
Comment (r0hitsharma) [inline nodejs/week2/app.js]: Please remember to set correct status codes for errors.

**Comment by Aditi-Sawardekar** [comment]
Thanks for reviewing my homework. I am working on suggestions.

**Comment by Aditi-Sawardekar** [comment]
Thanks for reviewing my homework.

File: nodejs/week2/app.js (hunk 2 above)
Comment (Aditi-Sawardekar) [reply nodejs/week2/app.js]: I worked again and  have now used regex to filter array.

File: nodejs/week2/app.js (hunk 1 above)
Comment (Aditi-Sawardekar) [reply nodejs/week2/app.js]: Changed code to use ES Modules.

File: nodejs/week2/app.js (hunk 3 above)
Comment (Aditi-Sawardekar) [reply nodejs/week2/app.js]: Made changes to handle if filtered items is empty

File: nodejs/week2/app.js (hunk 4 above)
Comment (Aditi-Sawardekar) [reply nodejs/week2/app.js]: Made changes to changes errors.

**Review by r0hitsharma — APPROVED** [review]
Improvements look good :+1:

**Comment by Aditi-Sawardekar** [comment]
Thank you Rohit for approving merge request.

**Pull request merged**